feat: jupyter free sessions in Renku v2 (#3712)
SwissDataScienceCenter/renku: CHANGELOG.rst

0.61.0
------

Renku 0.61.0 introduces a new version of Amalthea that supports running sessions with Docker images
that do not contain Jupyter server.

NOTE to administrators: This upgrade introduces a brand new CRD for sessions. All services that support
sessions for Renku v2 will switch to this new CRD. Renku v1 sessions remain unchanged.
Therefore any old sessions for Renku v2 will not be visible to users after this upgrade. The sessions
themselves will not be immediately deleted and as long as users have saved links to their old sessions they
should be able to access their sessions and save data. However we recommend that administrators
notify users of the change and allow for enough time so that existing Renku v2 sessions can be saved and
cleaned up, rather than asking users to save the url to their sessions. In addition to users not being able
to see old Renku v2 sessions, they will also not be able to pause, resume or delete old Renku v2 sessions.

> Therefore it best is most sessions are properly saved and cleaned up before this update is rolled out. In order 

Reviewer: ```suggestion
Therefore it's best if most sessions are properly saved and cleaned up before this update is rolled out. In order 
```

to support the new CRD we have also created a new operator that will manage the new `amaltheasession` resources.

User-Facing Changes
~~~~~~~~~~~~~~~~~~~

**✨ Improvements**

- **UI**: Enable the use of custom images that don’t contain Jupyter, streamlining the image-building process and allowing for the use of “off-the-shelf” images (`#3341 <https://github.com/SwissDataScienceCenter/renku-ui/pull/3341>`__).

> - **Sessions**: Enable running session images that do not container Jupyter in them.

Reviewer: ```suggestion
- **Sessions**: Enable running session images that do not contain Jupyter in them.
```

> - **Amalthea**: Brand new operator for sessions with images that do not have to container Jupyter

Reviewer: ```suggestion
- **Amalthea**: Brand new operator for sessions with images that do not have to contain Jupyter
```

Internal Changes
~~~~~~~~~~~~~~~~

**Improvements**

- **Data services**: Add support for OAuth storage providers
- **Data services**: Move notebooks code to data services

**Bug Fixes**

- **Gateway**: Pass on session cookie to data services for anonymous session authentication
- **Data services**: Correct pagination for namespaces

> - **Data services**: Add creation data and created_by for namespaces

Reviewer: ```suggestion
- **Data services**: Add creation date and created_by for namespaces
```
